lib/model.js

```diff
@@ -1786,9 +1786,21 @@ Model.prototype.bulkBuild = function(valueSets, options) { // testhint options:n
     });
   }
 
-  return valueSets.map(function(values) {
-    return this.build(values, options);
-  }.bind(this));
+  //if we requested a promise then we want to break up the event loop here by injecting some setImmediates
+  if(options.returnPromise) {
```

Comment:
Eventually it would be nice to refactor this entire chain of recursion to be promise based but its nested to deeply to make this as trivial as this change which addresses the loop locking pretty well.